doc/man7/EVP_PKEY-DSA.pod

```diff
@@ -10,9 +10,60 @@ EVP_PKEY-DSA, EVP_KEYMGMT-DSA, EVP_PKEY-DH, EVP_KEYMGMT-DH
 The B<DSA> and B<DH> keytypes are implemented in OpenSSL's default and FIPS
 providers.
 The implementations support the basic DSA and DH keys, containing the public
-and private keys I<pub> and I<priv> as well as the three domain parameters
+and private keys I<pub> and I<priv> as well as the three main domain parameters
 I<p>, I<q> and I<g>.
 
+Finite field cryptography (FFC) is a method of implementing discrete logarithm
+cryptography using finite field mathematics. DSA is an example of FFC and
+Diffie-Hellman key establishment algorithms specified in SP800-56A can also be
+implemented as FFC.
+
+For B<DH> FFC key agreement two classes of domain parameters can be used:
+“safe” domain parameters that are associated with approved named safe-prime
+groups, and a class of “FIPS 186-type” domain parameters. FIPS 186-type domain
+parameters should only be used for backward compatibility with existing
+applications that cannot be upgraded to use the approved safe-prime groups.
+
+For B<DSA> (and B<DH> that is not a named group) the FIPS186-4 standard
+specifies that the values used for FFC parameter generation are also required
+for parameter validation.
+This means that optional FFC domain parameter values for I<seed>, I<pcounter>
+and I<gindex> may need to be stored for validation purposes.
+For B<DH> the I<seed> and I<pcounter> can be stored in ASN1 data
+(but the I<gindex> is not). For B<DSA> however these fields are not stored in
+the ASN1 data so they need to be stored externally if validation is required.
+
+=head2 Common DH parameters
+
+=over 4
+
+=item "group" (B<OSSL_PKEY_PARAM_FFC_GROUP>) <UTF8 string>
+
+A string that associates a B<DH> named safe prime group with known values for
+I<p>, I<q> and I<g>.
+
+The following values can be used by OpenSSL's default and FIPS providers:
```

Comment:
Suggestion: `s/OpenSSL's/the/`

Reply:
Why?

Reply:
Note: I can accept "the default", but not "the FIPS"

Reply:
I meant: _The following values can be used by the default and FIPS providers_